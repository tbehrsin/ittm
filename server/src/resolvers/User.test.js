import { gql } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { server } from '../index';

export const GET_USERS_QUERY = gql`
  query getUsers {
    users {
      id
      email
      region
      blocked
      status
      createdAt
      role {
        id
        name
      }
    }
  }
`;

export const GET_USER_QUERY = gql`
  query getUser($id: Int!) {
    user(id: $id) {
      id
      email
      region
      blocked
      status
      createdAt
      role {
        id
        name
      }
    }
  }
`;

describe('User', () => {
  test('#users', async () => {
    const { query } = createTestClient(server);
    const { data } = await query({
      query: GET_USERS_QUERY,
    });
    expect(data).toMatchSnapshot();
  });

  test('#user', async () => {
    const { query } = createTestClient(server);
    const { data } = await query({
      query: GET_USER_QUERY,
      variables: {
        id: 456,
      },
    });
    expect(data).toMatchSnapshot();
  });

  test('#user not found', async () => {
    const { query } = createTestClient(server);
    const { data } = await query({
      query: GET_USER_QUERY,
      variables: {
        id: 0,
      },
    });
    expect(data).toMatchSnapshot();
  });
});
