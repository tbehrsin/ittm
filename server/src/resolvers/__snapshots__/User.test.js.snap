// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User #user 1`] = `
Object {
  "user": Object {
    "blocked": false,
    "createdAt": "2020-04-06T11:07:42.549Z",
    "email": "yolo.swaggins@example.com",
    "id": "456",
    "region": "UK",
    "role": Object {
      "id": "1",
      "name": "test role",
    },
    "status": "PENDING",
  },
}
`;

exports[`User #user not found 1`] = `
Object {
  "user": null,
}
`;

exports[`User #users 1`] = `
Object {
  "users": Array [
    Object {
      "blocked": false,
      "createdAt": "2020-04-06T11:07:42.549Z",
      "email": "support_admin@example.com",
      "id": "123",
      "region": "UK",
      "role": Object {
        "id": "1",
        "name": "test role",
      },
      "status": "ACTIVE",
    },
    Object {
      "blocked": false,
      "createdAt": "2020-04-06T11:07:42.549Z",
      "email": "yolo.swaggins@example.com",
      "id": "456",
      "region": "UK",
      "role": Object {
        "id": "1",
        "name": "test role",
      },
      "status": "PENDING",
    },
  ],
}
`;
